src/viur/core/modules/user.py

```diff
@@ -356,20 +363,11 @@ def login(self, *, name: str | None = None, password: str | None = None, **kwarg
         if iterations < PBKDF2_DEFAULT_ITERATIONS:
             logging.info(f"Update password hash for user {name}.")
             # re-hash the password with more iterations
-            skel = self._user_module.editSkel()
-            skel.setEntity(user_entry)
-            skel["key"] = user_entry.key
-            skel["password"] = password  # will be hashed on serialize
-            skel.toDB(update_relations=False)
+            # FIXME: This must be done within a transaction!
```

Comment:
```suggestion
```
`skel.toDB` always runs in a transaction or what doyou mean?

Reply:
Hello @ArneGudermann, yes you're right, the toDB() itself runs in a transaction, but it is necessary here to do the fromDB(), password assignment and toDB() within the transaction. It might overwrite already changed data with old data in this case. It's rare, but it can happen.

Reply:
Okay thats a point.